src/components/todo-item/todo-item.component.tsx

```diff
@@ -13,14 +15,27 @@ interface ITodoItemProps {
 const TodoItem = ({ todo, onChecked, onDelete }: ITodoItemProps) => {
     const { id, text, checked } = todo;
     const handleOnChange = () => { onChecked(id) }
-
+    const [color, setColor] = useState('#fff')
+    const [showColorPicker, setShowColorPicker] = useState(false)
     return (<li className={'todo-item'}>
         <input
             type="checkbox"
             checked={checked}
             onChange={handleOnChange}
         />
+
         <span className={`${checked ? 'checked' : ''}`}>{text}</span>
+
+
+                <BlockPicker
+                    color={color}
+                    onChange={updatedColor => setColor(updatedColor.hex)}
```

Comment:
we should not use callback functions like that.

you should extract function to const
like:

const updateColor = ({ hex }) => setColor(hex);

and then onChange={updateColor}

this is due to performance issue